x-pack/legacy/plugins/reporting/server/lib/create_worker.ts

```diff
@@ -39,17 +41,23 @@ function createWorkerFn(server: ServerFacade) {
       jobExecutors.set(exportType.jobType, executeJobFactory);
     }
 
-    const workerFn = (job: JobSource, jobdoc: JobDocPayload | JobDoc, cancellationToken?: any) => {
+    const workerFn = (
+      job: JobSource,
+      arg1: JobDocPayload | JobDoc,
```

Comment:
Maybe:

`arg1` => `job`
`arg2` => `cancelToken`?

Reply:
This is not a great part of the code - without Typescript, some bad design problems were hidden.

The idea is "worker functions" can come from different export types, but one export type changes the idea about what a worker function could be, that's the "immediate" execute type, which is in CSV export from Panel Action and doesn't go through job queuing. It registers a worker function that has a different signature. It'll be great to figure this out better when ESQueue goes away :)

Reply:
I'm going to leave this as-is because the arguments don't have a single semantic meaning